crates/holochain_p2p/src/spawn/actor.rs

```diff
@@ -199,6 +200,50 @@ impl HolochainP2pActor {
 impl ghost_actor::GhostHandler<kitsune_p2p::event::KitsuneP2pEvent> for HolochainP2pActor {}
 
 impl kitsune_p2p::event::KitsuneP2pEventHandler for HolochainP2pActor {
+    /// We need to store signed agent info.
+    fn handle_put_agent_info_signed(
+        &mut self,
+        input: kitsune_p2p::event::PutAgentInfoSignedEvt,
+    ) -> kitsune_p2p::event::KitsuneP2pEventHandlerResult<()> {
+        let kitsune_p2p::event::PutAgentInfoSignedEvt {
+            space,
+            agent,
+            agent_info_signed,
+        } = input;
+        let space = DnaHash::from_kitsune(&space);
+        let agent = AgentPubKey::from_kitsune(&agent);
+        let evt_sender = self.evt_sender.clone();
+        Ok(async move {
+            Ok(evt_sender
+                .put_agent_info_signed(space, agent, agent_info_signed)
+                .await?)
+        }
+        .boxed()
+        .into())
+    }
+
+    /// We need to get previously stored agent info.
+    fn handle_get_agent_info_signed(
+        &mut self,
+        input: kitsune_p2p::event::GetAgentInfoSignedEvt,
+    ) -> kitsune_p2p::event::KitsuneP2pEventHandlerResult<Option<AgentInfoSigned>> {
+        let kitsune_p2p::event::GetAgentInfoSignedEvt {
+            space,
+            agent,
+            agent_info_signed_key,
+        } = input;
```

Comment:
Here's where we could generate the AgentInfoSignedKey using the kitsune space/agent before they are transformed into AgentPubKey/DnaHash.